### ISSUE TITLE
NRT doesn't detect nullability of field that hasn't been declared yet

### ISSUE THREAD

**Issue by chucker** [dotnet/roslyn#49696]
Consider the following code:

```csharp
#nullable enable
    
private class NullBecauseOfDeclarationOrder
{
    private static readonly string S1 = "Value A";
    private static readonly string[] Strings1 = new string[] { S1 };

    private static readonly string[] Strings2 = new string[] { S2 };
    private static readonly string S2 = "Value B";
}
```

At compile time, this doesn't produce any warnings. However, at _runtime_, `Strings2` will contain `null` rathre than `Value B`.

That's because the produced IL actually _declares_ each field first, _then_ initializes them. When initializing `Strings2`, the value of `S2` is still `null`, so that gets inserted into the array.

However, nullable reference types don't seem to discover this as a problem. The compiler insists inside the array that `S2` is not null here, because it sees that

1. S2 is not marked as nullable, and
2. it is immediately set to a value

I believe this should yield either CS8618, or something similar:

> Warning	CS8618	Non-nullable field 'S2' is uninitialized. Consider declaring the field as nullable.

(A more correct warning would be "…is not yet initialized when accessed here".)
